File: openapi/java/impl/src/main/java/org/taymyr/lagom/demo/impl/PetsServiceImpl.java
package org.taymyr.lagom.demo.impl;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import org.taymyr.lagom.demo.api.NewPet;
import org.taymyr.lagom.demo.api.Pet;
import org.taymyr.lagom.demo.api.PetsService;

import java.util.List;
import java.util.Optional;

/**
 * Implementation of the PetsService.
 */
public class PetsServiceImpl implements PetsService {

    @Override
    public ServiceCall<NotUsed, List<Pet>> find(List<String> tags, Optional<Integer> limit) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public ServiceCall<NewPet, Pet> create() {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public ServiceCall<NotUsed, Pet> findBy(Long id) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public ServiceCall<NotUsed, NotUsed> delete(Long id) {
        throw new UnsupportedOperationException("Not implemented");
    }

}

File: openapi/scala/impl/src/main/scala/org/taymyr/lagom/demo/impl/PetsServiceImpl.scala
package org.taymyr.lagom.demo.impl

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.ServiceCall
import com.typesafe.config.Config
import org.taymyr.lagom.demo.api.{NewPet, Pet, PetsService}
import org.taymyr.lagom.scaladsl.openapi.OpenAPIServiceImpl

/**
  * Implementation of the PetsService.
  */
class PetsServiceImpl extends PetsService {
  override def find(tags: List[String],
                    limit: Option[Int]): ServiceCall[NotUsed, List[Pet]] = ???

  override def create: ServiceCall[NewPet, Pet] = ???

  override def findBy(id: Long): ServiceCall[NotUsed, Pet] = ???

  override def delete(id: Long): ServiceCall[NotUsed, NotUsed] = ???
}
